sdks/go/pkg/beam/core/sdf/sdf.go

```diff
@@ -73,8 +73,8 @@ type RTracker interface {
 	// reason), then this function returns nil as the residual.
 	//
 	// If the split fraction is 0 (e.g. a self-checkpointing split) TrySplit() should return either
-	// a nil primary or an RTracker that is both bounded and has size 0. This ensures that there is
-	// no data that is lost by not being rescheduled for execution later.
+	// a nil primary or a restriction that represents no remaining work. This will ensure that there
+	// is not data loss.
```

Comment:
Really we don't care about the primary that they return, right? The thing we really care about is that they've correctly set their RTracker such that IsDone() returns true.

Also, the current consequence isn't data loss, its a failed pipeline I think

Reply:
> So the returned primary restriction is supposed to be the same as the restriction the RTracker holds as part of a typical runner-side split, so in general we do care that the returns are configured correctly.

That's true, but if somebody is debugging their pipeline the IsDone is the important bit. I don't mind saying both, but I think it would be good to explicitly call that out.

Reply:
That's fair. I've added callouts to TrySplit() and IsDone() to try and make the behavior clear for users.